### ISSUE TITLE
pre-installed pip complains it is out of date

### ISSUE THREAD

**Issue by jayvdb** [ogrisel/python-appveyor-demo#15]
As the pre-installed version version of pip is typically out of date soon, and install.ps1 does not upgrade it, pip reports a warning on stderr

```
pip : You are using pip version 6.1.1, however version 7.0.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

As it is on stderr, it does not play nicely with using pip in PowerShell, which believes that is an exception

```
pip : You are using pip version 6.1.1, however version 7.0.3 is available.
At line:1 char:33
+ if($env:COVERALLS_REPO_TOKEN) { pip install coverage }
+                                 ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (You are using p...3 is available.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
You should consider upgrading via the 'pip install --upgrade pip' command.
```

https://ci.appveyor.com/project/jayvdb/python-appveyor-demo/build/1.0.25/job/ngtpnkqguoumr7uo

**Comment by ogrisel** [comment]
Let me know if 429f949a44d8241240645b93c45ba2ca884e3a2e fixes this problem.

**Comment by jayvdb** [comment]
Big bloody boom on all the pre-install environments: https://ci.appveyor.com/project/ogrisel/python-appveyor-demo/build/1.0.127

Unfortunately it needs to be done more carefully, in install.ps1 probably, to avoid permission problems with the pre-installed package.  Or we might need to ask Appveyor to upgrade their pip regularly.

**Comment by ogrisel** [comment]
Argl. Will revert. How doing it in `install.ps1` would make it possible to workaround the permission problems?

**Comment by jayvdb** [comment]
One idea is to set up a virtualenv (or similar) for the pre-installed environments.

**Comment by ogrisel** [comment]
That could work indeed. Would you like to work on that?

**Comment by jayvdb** [comment]
It looks like this problem has been solved by Appveyor, for the pre-installed Python 3.3 and Python 3.4.
https://ci.appveyor.com/project/jayvdb/python-appveyor-demo/build/1.0.83/job/69tg8avv01bwcmp5#L65
https://ci.appveyor.com/project/jayvdb/python-appveyor-demo/build/1.0.83/job/der1qg5358coh3bx#L66

However it isnt solved for Python versions that come with pip in the installer; e.g. 2.7.10 comes with 7.0.1, and the latest is 7.1.0
https://ci.appveyor.com/project/jayvdb/python-appveyor-demo/build/1.0.83/job/a8qasty25fwb5pxk#L78

@VcamX has worked on using virtualenv.  You can see the proposed changes at
https://gerrit.wikimedia.org/r/#/c/227239/3/.appveyor.yml,cm

However, a virtualenv isnt needed.  `pip install --user --upgrade pip` solves this problem.
It looks like the bug encountered with 429f949a is https://github.com/pypa/pip/issues/2863

**Comment by jayvdb** [comment]
This doesnt appear to be a problem with the pip which is distributed in the Windows installer for Python 3.4.  the pip feature to detect when it is out of date was introduced in pip 6.0 .  However only pip 1.5.x is included in Python 3.4.x installers.

Python 3.4.0 - pip 1.5.4
Python 3.4.1 & Python 3.4.2 - pip 1.5.6

So, this problem is currently limited to the Python 2.7.10 installer, which includes the backport of the pip pre-installer, but comes with pip 7.0.1.
